src/analyzer/plugins/analyzerqueenmarybeats.cpp

```diff
@@ -39,9 +47,8 @@ AnalyzerQueenMaryBeats::~AnalyzerQueenMaryBeats() {
 bool AnalyzerQueenMaryBeats::initialize(int samplerate) {
     m_detectionResults.clear();
     m_iSampleRate = samplerate;
-    // These are the preferred window/step sizes from the BeatTrack VAMP
-    m_stepSize = int(m_iSampleRate * kStepSecs + 0.0001);
-    m_windowSize = m_stepSize * 2;
+    m_stepSize = m_iSampleRate * kStepSecs;
```

Comment:
Why do we still use this odd constant kStepSecs that results exactly in m_stepSize = m_windowSize / 2 at 44100 kHz but not at other sample rates? Why not always use m_windowSize / 2? I am not familiar with the math but kStepSecs seems arbitrary.

Reply:
> The onset detection function is calculated by measuring the spectral difference between short term (11 ms) analysis frames and is referred to as the complex domain method

10 ms is considered a good value for computing onset detection, I suspect that the rather arbitrary value of this constant was chosen exactly to make the step size equals to 512 (and thus windows size 1024) in 44.1khz.

I have experimented with rounding the stepsize to the closest power of 2 before and keep the window size the double of the step size, which is the same as what you are proposing, but Daniel's approach seems to work better. ie: the difference in results of the same track in different sample rates was more noticeable.

Reply:
The old code gives the impression that step size and window size is related.
That is wrong. 
The window size sets the number of frequency bins. And the step size is just the sample rate. Of the resulting frequency domain. Both can be adjusted independent.

Reply:
I am not sure but I think there is a reasoning for the perfect overlapping

Reply:
The overlapping of 1/2 window length makes sense in a different context.

It ensures that every sample is treated the same. Border samples in the first window become center samples in the following. 

This is important if everage values are from interest, or if it is desired to reconstruct the audio from the fft bins.

I think this is not the case here. But prove me wrong ..